shell: Update exit code for special cases

chu11 opened the pull request:
Per discussion in #2752, quick fix to handle special case exit codes in the job shell.

grondo commented on src/shell/shell.c:
```diff
@@ -1114,7 +1114,17 @@ int main (int argc, char *argv[])
+                ec = 126;
+            else if (errno == ENOENT)
+                ec = 127;
+            else if (errno == EHOSTUNREACH)
```
Since the shell is only launching local tasks, can we ever get `EHOSTUNREACH` from shell_task_start?
(I'm probably just missing a corner case, since you probably saw this errno during testing)

chu11 replied:
> (I'm probably just missing a corner case, since you probably saw this errno during testing)

You think too highly of me :-)  I blindly copied behavior from other parts of Flux.  You're absolutely right, the shell only does local tasks.

grondo commented on t/t2608-job-shell-log.t:
```diff
@@ -133,12 +133,20 @@ for topic in "shell.init" "shell.exit" \
+	test_expect_code 126 flux mini run adirectory 2>illegal.err &&
+	grep "flux-shell\[0\]: FATAL: task 0: start failed" illegal.err &&
+	grep "job.exception type=exec severity=0 task 0: start failed" illegal.err &&
+        grep "Permission denied" illegal.err
```
nit: looks like maybe a couple indentation errors in this test?

chu11 replied:
Yup, there was a tab vs space issue.  Went through the file and it seems we got a lot of tab vs space inconsistencies, and in other tests.  Maybe should do a big cleanup one of these days.

chu11 commented:
re-pushed, removing EHOSTUNREACH and tweaking some tabbing per comments above.  There are more tabbing inconsistencies in `t2608-job-shell-log.t`, but those fixes are probably for another PR.

grondo commented:
>  There are more tabbing inconsistencies in t2608-job-shell-log.t, but those fixes are probably for another PR.

Yeah, sorry that must be my fault...

chu11 commented:
> Yeah, sorry that must be my fault...

Skimming through the tests, the inconsistencies are everywhere, so I'm sure it was a collective effort! :-)

chu11 commented:
Hmmm, almost all builders failed:

```
expecting success: 
	mkdir adirectory &&
	test_expect_code 126 flux mini run adirectory 2>illegal.err &&
	grep "flux-shell\[0\]: FATAL: task 0: start failed" illegal.err &&
	grep "job.exception type=exec severity=0 task 0: start failed" illegal.err &&
	grep "Permission denied" illegal.err

not ok 132 - flux-shell: illegal command logs fatal error
```

Nothing on stderr, so mkdir failed?  Perhaps I need to `chdir` to temp or something for travis.

Removing merge-when-passing so I can add some debugging commits

chu11 commented:
hmmm

```
expecting success: 
	mkdir adirectory &&
flux mini run adirectory &&
	test_expect_code 126 flux mini run adirectory 2>illegal.err &&
	grep "flux-shell\[0\]: FATAL: task 0: start failed" illegal.err &&
	grep "job.exception type=exec severity=0 task 0: start failed" illegal.err &&
	grep "Permission denied" illegal.err

0.076s: flux-shell[0]: FATAL: task 0: start failed: adirectory: No such file or directory
0.077s: job.exception type=exec severity=0 task 0: start failed: adirectory: No such file or directory
flux-job: task(s) exited with exit code 127
not ok 132 - flux-shell: illegal command logs fatal error
```

I'm getting "No such file or directory", but the `mkdir adirectory` clearly succeeds on the line above.  `umask` issue? hmmm

chu11 commented:
Guess it was a PATH issue, this fixes things:

```
-       test_expect_code 126 flux mini run adirectory 2>illegal.err &&
+       test_expect_code 126 flux mini run ./adirectory 2>illegal.err &&
```

squashed, rebased, and re-pushed.  I don't think I should reset `merge-when-passing`, so if someone else can, thanks!

grondo commented:
One quick thing -- the commit message still mentions EHOSTUNREACH

chu11 commented:
> One quick thing -- the commit message still mentions EHOSTUNREACH

Doh! Thanks, repushed

grondo commented:
Nice! Re-setting merge-when-passing.
